util/src/main/java/com/coderanch/util/matchers/Matchers.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2018 Coderanch.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.coderanch.util.matchers;
+
+import java.util.function.*;
+
+import org.hamcrest.CustomTypeSafeMatcher;
+import org.hamcrest.Matcher;
+
+import static org.hamcrest.Matchers.both;
+import static org.hamcrest.Matchers.isEmptyString;
+import static org.hamcrest.Matchers.not;
+
+/**
+ * Utility class to create common matchers.
+ */
+public final class Matchers {
+
+    private Matchers() {}
+
+    private static final Matcher<String> TRIMMED_STRING_MATCHER = createTypeSafeMatcher(
+        "a trimmed string",
+        string -> string.trim().equals(string)
+    );
+
+    /**
+     * Creates a type-safe matcher from a description and a predicate.
+     *
+     * @param <T>         the type of the argument that will be matched.
+     *
+     * @param description a description of the argument that will be matched.
+     * @param predicate   a predicate that will be applied to the argument when it is matched.
+     *
+     * @return a matcher with the given {@code description} that will successfully match an argument
+	 *         if the {@code predicate} on that argument returns {@code true}.
+     */
+    public static <T> Matcher<T> createTypeSafeMatcher(String description, Predicate<? super T> predicate) {
+        return new CustomTypeSafeMatcher<T>(description) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Reduce this anonymous class number of lines from 6 to at most 5, or make it a named class. ([squid:S1188](https://sonarcloud.io/organizations/coderanch-corral/rules#rule_key=squid%3AS1188))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.coderanch%3Ablackjack-parent&pullRequest=11&issues=AWaB526VksgJibV5mvGA&open=AWaB526VksgJibV5mvGA)

[ISSUE_KEY]: AWaB526VksgJibV5mvGA